### DISCUSSION TITLE
Custom Controls Styling?

### DISCUSSION THREAD

**Discussion by kcapka** [cookpete/react-player#1852] [Q&A] [answered]
Hello! I am trying to create custom controls for react-player that allow for an identical user experience no matter where the video comes from (YouTube, Vimeo, etc.). I understand how I can achieve the functionality, but due to the structure of ReactPlayer, I can't seem to get the controls to position themselves over top of the video.

In the react-player docs, there is a brief mention to adding custom players. ReactPlayer.addCustomPlayer() is mentioned, but I cannot find any more info as to what a custom player is or what format it should be developed in.

Any help/advice would be appreciated!

**Comment by luwes** [comment, marked as answer]
thanks for the feedback! 

the `ReactPlayer.addCustomPlayer()` method is for adding new players of other platforms (think Spotify, Cloudflare, etc.), it's not related to custom controls.

we do have this feature in mind and would like to support https://github.com/muxinc/media-chrome/ in the v3.
https://github.com/cookpete/react-player/discussions/1843

Until then I'd recommend having a look at media-chrome and the supported media-elements https://github.com/muxinc/media-elements
for a uniform custom player UI and different video providers.

**Reply by kcapka** [reply]
Great thank you so much!

**Reply by kcapka** [reply]
@luwes I was able to construct custom controls and get them working well using the react-player props, but that has brought me to one final question. I don't see any listed props to trigger full-screen without enabling the native controls. Are there any plans to try to implement this in the future?
